src/use-swr.ts

```diff
@@ -601,6 +604,42 @@ function useSWR<Data = any, Error = any>(
     revalidate
   ])
 
+  // setup subscription
+  useEffect(() => {
+    // If we haven't received a subscribe function we will do nothing here
+    if (!config.subscribe) return undefined
+
+    // If it's the first time a key will subscribe we will define the key
+    if (!SUBSCRIBERS[key]) {
+      SUBSCRIBERS[key] = { count: 0, unsubscribe: () => {} }
+    }
+
+    // If at the moment we run this there are no other subscribers for this key
+    if (SUBSCRIBERS[key].count === 0) {
+      // we will increment it by one
+      SUBSCRIBERS[key].count += 1
+      // and create the subscription storing the unsubscribe key
+      SUBSCRIBERS[key].unsubscribe = config.subscribe(_key, boundMutate)
```

Comment:
Why `_key`? If we are using a function key (`useSWR(() => throwOrReturnKey)`) I feel it's better to have the key value (with args) here, wdyt?